flexprep/services/processing_service.py

```diff
@@ -0,0 +1,38 @@
+"""
+Services providing core functionality.
+"""
+
+import logging
+
+from flexprep import CONFIG
+from flexprep.domain.prepare_processing import PrepProcessing
+from flexprep.domain.s3_utils import S3client
+
+logger = logging.getLogger(__name__)
+
+
+def _check_s3input_content(s3_client, bucket_name: str) -> bool:
+    try:
+        s3_objects = s3_client.list_objects_v2(Bucket=bucket_name)
+        if "Contents" not in s3_objects:
+            logger.error(f"No objects found in bucket {bucket_name}")
+            return False
+        return True
+    except Exception as e:
+        logger.error(f"Error checking S3 bucket content: {e}")
+        return False
+
+
+def process():
+    logger.debug("Initialize Processing")
+
+    s3_input = S3client().s3_client_input
+    bucket_name = CONFIG.main.s3_buckets.input.name
+
+    if not _check_s3input_content(s3_input, bucket_name):
+        logger.error("S3 input content check failed. Aborting processing.")
+        return
```

Comment:
I would raise an exception rather than returning. Seen in a few places than you handle an exception by logging the error and returning. I think better to log + re-raise